translate_test.ipynb

```diff
@@ -0,0 +1,170 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "id": "initial_id",
+   "metadata": {
+    "collapsed": true,
+    "jupyter": {
+     "is_executing": true
+    }
+   },
+   "source": [
+    "\n",
+    "import os\n",
+    "from dotenv import load_dotenv\n",
+    "from libs.community.langchain_community.tools.azure_ai_services.translate_tool import AzureTranslateTool\n",
+    "from langchain_core.prompts import PromptTemplate\n",
+    "\n",
+    "load_dotenv()"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-10-10T02:47:56.337282Z",
+     "start_time": "2024-10-10T02:47:56.322998Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "translate_key = os.getenv('AZURE_TRANSLATE_API_KEY')\n",
+    "translate_endpoint = os.getenv('AZURE_TRANSLATE_ENDPOINT')\n"
+   ],
+   "id": "d16a2afc75afc1a4",
+   "outputs": [],
+   "execution_count": 9
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-10-10T02:47:56.353227Z",
+     "start_time": "2024-10-10T02:47:56.338283Z"
+    }
+   },
+   "cell_type": "code",
+   "source": "translator = AzureTranslateTool(translate_key=translate_key, translate_endpoint=translate_endpoint)",
+   "id": "6a9a508bab82e582",
+   "outputs": [],
+   "execution_count": 10
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-10-10T02:47:56.368993Z",
+     "start_time": "2024-10-10T02:47:56.355229Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "\n",
+    "prompt = PromptTemplate(\n",
+    "    input_variables=['input'],\n",
+    "    template=\"You are a translation agent. Use the translation tool to translate the following input: {input}\"\n",
+    ")\n"
+   ],
+   "id": "e4d08ff8441d3ac1",
+   "outputs": [],
+   "execution_count": 11
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-10-10T02:47:56.384562Z",
+     "start_time": "2024-10-10T02:47:56.369994Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "def translate_text(input_text, target_language='es'):\n",
+    "    try:\n",
+    "        # Run the translation\n",
+    "        translated_text = translator._run(input_text, target_language)\n",
```

Comment:
We shouldn't call the internal methods. I'd expect a user to invoke the tool directly or use an agent.

```python
translate_tool = AzureTranslateTool(
    text_translation_key=os.environ["AZURE_TRANSLATE_API_KEY"],
    text_translation_endpoint=os.environ["AZURE_TRANSLATE_ENDPOINT"],
    region="westus2",
)
answer = translate_tool.invoke({"query": "hola mundo"})

print(answer)
```

```python
translate_tool = AzureTranslateTool(
    text_translation_key=os.environ["AZURE_TRANSLATE_API_KEY"],
    text_translation_endpoint=os.environ["AZURE_TRANSLATE_ENDPOINT"],
    region="westus2",
)

tools = [translate_tool]

model = AzureChatOpenAI(
    openai_api_version=os.environ["OPENAI_API_VERSION"],
    azure_deployment=os.environ["COMPLETIONS_MODEL"],
    azure_endpoint=os.environ["AZURE_OPENAI_ENDPOINT"],
    api_key=os.environ["AZURE_OPENAI_API_KEY"],
)

agent = create_structured_chat_agent(model, tools, prompt)

agent_executor = AgentExecutor(
    agent=agent, tools=tools, verbose=True, handle_parsing_errors=True
)

text = "hola mundo"

answer = agent_executor.invoke(
    {"input": f"What does the text say in english? : {text}"}
)
print(answer)
```